S32-io/file-tests.t

```diff
@@ -3,71 +3,249 @@ use Test;
 
 # L<S32::IO/IO::FileTests>
 
-plan 35;
+plan 7;
 
-my $existing-file = "tempfile-file-tests";
-my $non-existent-file = "non-existent-file-tests";
-my $zero-length-file = "tempfile-zero-length-file-tests";
+my %tempfiles =
+    existing => "tempfile-file-tests",
+    non-existing => "non-existent-file-tests",
+    zero => "tempfile-zero-length-file-tests",
+    symlink-existing => "symlink-existing",
+    symlink-non-existing => "symlink-nonexisting",
+    r => "tempfile-perms-r",
+    w => "tempfile-perms-w",
+    x => "tempfile-perms-x",
+    rw => "tempfile-perms-rw",
+    rwx => "tempfile-perms-rwx",
+    ;
 
 { # write the file first
-    my $fh = open($existing-file, :w);
+    my $fh = open(%tempfiles<existing>, :w);
     $fh.print: "0123456789A";
     $fh.close();
 }
 
 { # write the file first
-    my $fh = open($zero-length-file, :w);
+    my $fh = open(%tempfiles<zero>, :w);
     $fh.close();
 }
 
+{ # Create a symlink to an existing file
+    symlink(%tempfiles<existing>, %tempfiles<symlink-existing>);
+}
```

Comment:
Instead of creating all of these manually in the current dir and then manually cleaning them up, you should add to the top of the file (after `use v6`):

```perl6
use lib <t/spec/packages/>;
use Test::Util;
```

And that will [give you `make-temp-file` and `make-temp-dir` routines](https://github.com/perl6/roast/blob/3014999fcddc4590338ffce842476da12a6817b8/packages/Test/Util.pm#L519-L534). `make-temp-file` without `:content(...)` argument creates just a temp path so you could use that for `non-existing` file too. Alternatively, you can just `chdir` to a `make-temp-dir`'ed dir and create all of these files up in there.

This way the test file will cut down on setup code all of these files aren't left hanging around in the current dir if the spectest is interrupted.